src/js/mediaStore.js

```diff
@@ -0,0 +1,108 @@
+      const { media } = stateOwners;
+      if (!media) return;
+      if (value) {
+        media.pause();
+      } else {
+        media.play().catch(() => {});
+      }
+    },
+    mediaEvents: ['play', 'pause', 'emptied'],
+  },
+};
+
+// Allow us to default/initialize our relevant state owners
+const createMediaStore = ({ media, fullscreenElement }) => {
+  const callbacks = [];
+
+  // We may eventually want to expose these as part of the state
+  // for advanced use cases
+  const stateOwners = {
+    media,
+  };
+
+  // In a real version, the initial state of this could likely just be a generic transform
+  // of the `stateFacade`
+  let state = Object.freeze({
+    mediaPaused: stateFacade.mediaPaused.get(stateOwners),
+  });
+
+  if (stateOwners.media) {
+    // This is similar to the `<media-controller>` setup for state monitoring
+    // and state propagation
+    Object.entries(stateFacade).forEach(([stateName, { get, mediaEvents }]) => {
+      // In a full version, we'd want to store these somewhere so
+      // we can remove if the media is removed/replaced
+      const handler = () => {
+        const nextValue = get(stateOwners);
+        if (state[stateName] === nextValue) return;
+
+        // Update the state since it changed.
+        // Using an "immutable" approach here so
+        // callbacks can easily do comparisons between prev/next state.
+        // Freezing isn't necessary, though it's a light touch enforcement
+        // of immutability (in case folks try to directly modify state)
+        state = Object.freeze({
+          ...state,
+          [stateName]: nextValue,
+        });
+
+        // Given anything that cares the updated state
+        callbacks.forEach((cb) => cb(state));
+      };
+      mediaEvents.forEach((eventType) => {
+        stateOwners.media.addEventListener(eventType, handler);
+      });
+    });
+  }
+
+  return {
+    // note these "actions" could just be the events if we wanted, especially if we normalize on "detail" for
+    // any payload-relevant values
+    dispatch({ type, detail }) {
+      if (
+        [
+          MediaUIEvents.MEDIA_PAUSE_REQUEST,
+          MediaUIEvents.MEDIA_PLAY_REQUEST,
+        ].includes(type)
+      ) {
+        // Most if not all state change requests can be modeled as UPDATE requests (in the generic "CRUD" sense)
+        // 1. Name of state to change (in this case, "paused")
+        // 2. Value you want the state to be (in this case, true if MEDIA_PAUSE_REQUEST, false if MEDIA_PLAY_REQUEST)
+        // We can think of more specific requests (e.g. MEDIA_PAUSE_REQUEST, MEDIA_PLAY_REQUEST) as just shorthand
+        // translations of e.g. { type: 'mediaupdaterequest', key: nameOfState, value: valueOfState }. In this case,
+        // 'mediapauserequest'|'mediaplayrequest' can be thought of as shorthand/convenience actions for something like:
+        // { type: 'mediaupdaterequest', detail: { key: 'mediaPaused', value: true | false } }
```

Comment:
I'm not seeing the value in this yet myself. While I do see complexity of then understanding that a `mediaplayrequest` gets translated into a paused update before then being translated back into a media.play function call. If we're gonna introduce those logic and mental jumps there's gotta be a really good reason for it.

Reply:
This is just a mental model that could also make the updates "generic" if we ever wanted to "pre-transform to the shape ☝️). That said, I'm not especially concerned about this one and we can do whatever folks find easier to work with/best when weighing the tradeoffs.